sips/sip-030/sip-030-wallet-interface.md

```diff
@@ -0,0 +1,571 @@
+
+To ensure serializability, consider these notes:
+
+- Enums are serialized as human-readable strings.
+- BigInts are serialized as numbers, strings, or anything that can be parsed by the JavaScript BigInt constructor.
+- Bytes are serialized as hex-encoded strings (without a 0x prefix).
+- Predefined formats from previous SIPs are used where applicable.
+- Addresses are serialized as Stacks c32-encoded strings.
+- Clarity values, post-conditions, and transactions are serialized to bytes (defined by SIP-005) and used as hex-encoded strings.
+
+### Methods
+
+This section defines the available methods, their parameters, and result structure.
+Parameters should be considered only as recommendations for the wallet, and the user/wallet may choose to ignore or override them.
+
+> Note: Optional params are marked with a `?`.
+
+Methods can be namespaced under `stx_` if used in more generic settings and other more Ethereum inspired domains.
+In other cases (e.g. `WalletConnect`), the namespace may already be given by metadata (e.g. a `chainId` field) and can be omitted.
+On the predominant `StacksProvider` global object, the methods can be used without a namespace, but wallets may add namespaced aliases for convenience.
+
+##### General parameters (for transaction methods)
+
+The following definitions can be used in the transaction methods.
+
+Parameter properties
+
+- `address?`: `string` address, Stacks c32-encoded, defaults to wallets current address
+- `network?`: `'mainnet' | 'testnet' | 'regtest' | 'devnet' | 'mocknet' | NetworkId`
+  - where `NetworkId extends string` and can be a network identifier, which already exists in the wallet (e.g. a URL or network name)
+- `fee?`: `number | string` BigInt constructor compatible value
+- `nonce?`: `number | string` BigInt constructor compatible value
+- `postConditions?`: `PostCondition[]`, defaults to `[]`
+  - where `PostCondition` is `string | object` hex-encoded or JSON representation
+- `postConditionMode?`: `'allow' | 'deny'`
+- `sponsored?`: `boolean`, defaults to `false`
+- ~~`attachment?`~~ _removed_
+- ~~`appDetails`~~ _removed_
+- ~~`onFinish`~~ _removed_
+- ~~`onCancel`~~ _removed_
+
+---
+
+#### Method `stx_transferStx`
+
+> **Comment**: This method doesn't take post-conditions.
+
+Parameter properties
+
+- `recipient`: `string` address, Stacks c32-encoded
+- `amount`: `number | string` BigInt constructor compatible value
+- `memo?`: `string`, defaults to `''`
+
+Result properties
+
+- `txid`: `string` hex-encoded
+- `transaction`: `string` hex-encoded raw transaction
+
+#### Method `stx_transferSip10Ft`
+
+Parameter properties
+
+- `recipient`: `string` address, Stacks c32-encoded
+- `asset`: `string` address, Stacks c32-encoded, with contract name suffix
+- `amount`: `number | string` BigInt constructor compatible value
+
+Result properties
+
+- `txid`: `string` hex-encoded
+- `transaction`: `string` hex-encoded raw transaction
+
+#### Method `stx_transferSip9Nft`
```

Comment:
stx_transferSip13Sft could be added as well.

Reply:
I can't tell if here SFTs should be treated as FTs

> Transfer
>
> (transfer ((token-id uint) (amount uint) (sender principal) (recipient principal)) (response bool uint))
> 
> Transfer a token from the sender to the recipient. It is recommended to leverage Clarity primitives like ft-transfer? to help safeguard users. The function should return (ok true) on success or an err response containing an unsigned integer on failure. The failure codes follow the existing conventions of stx-transfer? and ft-transfer?.

Reply:
SFTs have one more parameter, therefore, it should be specified as well.

Reply:
@janniks any comment on this. Last comment to resolve before merging.